### ISSUE TITLE
Local package.json exists, but node_modules missing

### ISSUE THREAD

**Issue by bh90210** [wailsapp/wails#92]
**Description**
I am following the guide found [here](https://wails.app/tutorials/template.html) from Wails' documentation. After running `wails init` -without first running `wails build` since it is not noted as mandatory- I cd in frontend directory and try `npm run serve`. I get thrown a warning `Local package.json exists, but node_modules missing, did you mean to install?` (see below). Running `npm install` "fixes" the problem.

**To Reproduce**
Steps to reproduce the behaviour:
1. `wails init`
2. `cd projectname`
3. `cd frontend`
4. `npm run serve`
5. error thrown
```
sh: 1: vue-cli-service: not found
npm ERR! file sh
npm ERR! code ELIFECYCLE
npm ERR! errno ENOENT
npm ERR! syscall spawn
npm ERR! testwails@ serve: `vue-cli-service serve`
npm ERR! spawn ENOENT
npm ERR! 
npm ERR! Failed at the testwails@ serve script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm WARN Local package.json exists, but node_modules missing, did you mean to install?
```

**Expected behaviour**
Serve the frontend

**Screenshots**
![error](https://user-images.githubusercontent.com/22690219/57972513-8c7b5080-79a4-11e9-82d4-9bf86eb4c65d.png)

**System Details**
Wails v0.14.4-pre
Ubuntu 18.04
go1.12.3
npm 6.9.0

**Comment by leaanthony** [comment]
Thanks for this. After init, i think it does say to cd to the directory and run `wails build` but maybe the tutorial doesn't say it. However, I'm going to consider this a bug and force serve to build the frontend if it hasn't been done. 👍

**Comment by BH90210** [comment]
yes, it says 
```
$ wails init
Wails v0.14.4-pre - Initialising project

The name of the project (My Project): testwailsinit
Project Name: testwailsinit
The output binary name (testwailsinit): 
Output binary Name: testwailsinit
Project directory name (testwailsinit): 
Project Directory: testwailsinit
Template: Vue2/Webpack Basic
Project 'testwailsinit' generated in directory 'testwailsinit'!
To compile the project, run 'wails build' in the project directory.
```
But I assumed that compile here is referring to a binary -*not to vue*- so I thought I don't need to run it immediately after `init` since I have `serve` available.

Also going through the documentation I couldn't find somewhere clearly stated that I should run `wails build` first. But in any way if the init command `wails init` incorporates the `npm install` there is no need to change anything in the documentation.

If the fix comes at `wails serve` level though I think there should be a clearer feedback to the user that before attempting to move inside frontend & running `npm run serve` on a newly created project, they should at least first run `wails serve` from main project dir.

This got my confused twice. The first time, a couple of weeks back I didn't pay attention, but going through it the second time I realised that is indeed confusing and worth reporting just for the time spent `npm install` things my self and continuous looping through documentation to find what I possibly could have done wrong.

![ref1](https://user-images.githubusercontent.com/22690219/57979205-3699bd80-7a23-11e9-95f3-273e1506377f.png)
![ref2](https://user-images.githubusercontent.com/22690219/57979206-3699bd80-7a23-11e9-9f12-4274d1600b56.png)
![ref3](https://user-images.githubusercontent.com/22690219/57979207-3699bd80-7a23-11e9-856b-8d949b02adf4.png)

**Comment by leaanthony** [comment]
Checkout https://github.com/wailsapp/wails/tree/92-build-on-first-init and see if that works better for you. The docs will need updating if you think this is a good solution.

**Comment by bh90210** [comment]
hmm so please bear with me on this one, cause I am not sure I do everything as I should.

1. I deleted $GOPATH/src/github.com/wailsapp/ (though I guess wasn't really necessary) and did a fresh `go get github.com/wailsapp/wails/cmd/wails`
2. cd $GOPATH/src/github.com/wailsapp/wails
3. git checkout --track origin/92-build-on-first-init
4. go get

then I get this error
```
$ go get
go get: warning: modules disabled by GO111MODULE=auto in GOPATH/src;
	ignoring go.mod;
	see 'go help modules'
# github.com/wailsapp/webview
In file included from webview.go:27:0,
                 from _cgo_export.c:4:
../webview/webview.h: In function ‘external_message_received_cb’:
../webview/webview.h:278:5: warning: ‘webkit_javascript_result_get_global_context’ is deprecated [-Wdeprecated-declarations]
     JSGlobalContextRef context = webkit_javascript_result_get_global_context(r);
     ^~~~~~~~~~~~~~~~~~
In file included from /usr/include/webkitgtk-4.0/webkit2/webkit2.h:55:0,
                 from ../webview/webview.h:45,
                 from webview.go:27,
                 from _cgo_export.c:4:
/usr/include/webkitgtk-4.0/webkit2/WebKitJavascriptResult.h:49:1: note: declared here
 webkit_javascript_result_get_global_context (WebKitJavascriptResult *js_result);
 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from webview.go:27:0,
                 from _cgo_export.c:4:
../webview/webview.h:279:5: warning: ‘webkit_javascript_result_get_value’ is deprecated: Use 'webkit_javascript_result_get_js_value' instead [-Wdeprecated-declarations]
     JSValueRef value = webkit_javascript_result_get_value(r);
     ^~~~~~~~~~
In file included from /usr/include/webkitgtk-4.0/webkit2/webkit2.h:55:0,
                 from ../webview/webview.h:45,
                 from webview.go:27,
                 from _cgo_export.c:4:
/usr/include/webkitgtk-4.0/webkit2/WebKitJavascriptResult.h:52:1: note: declared here
 webkit_javascript_result_get_value          (WebKitJavascriptResult *js_result);
 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# github.com/wailsapp/webview
In file included from ../webview/webview.go:27:0:
../webview/webview.h: In function ‘external_message_received_cb’:
../webview/webview.h:278:5: warning: ‘webkit_javascript_result_get_global_context’ is deprecated [-Wdeprecated-declarations]
     JSGlobalContextRef context = webkit_javascript_result_get_global_context(r);
     ^~~~~~~~~~~~~~~~~~
In file included from /usr/include/webkitgtk-4.0/webkit2/webkit2.h:55:0,
                 from ../webview/webview.h:45,
                 from ../webview/webview.go:27:
/usr/include/webkitgtk-4.0/webkit2/WebKitJavascriptResult.h:49:1: note: declared here
 webkit_javascript_result_get_global_context (WebKitJavascriptResult *js_result);
 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from ../webview/webview.go:27:0:
../webview/webview.h:279:5: warning: ‘webkit_javascript_result_get_value’ is deprecated: Use 'webkit_javascript_result_get_js_value' instead [-Wdeprecated-declarations]
     JSValueRef value = webkit_javascript_result_get_value(r);
     ^~~~~~~~~~
In file included from /usr/include/webkitgtk-4.0/webkit2/webkit2.h:55:0,
                 from ../webview/webview.h:45,
                 from ../webview/webview.go:27:
/usr/include/webkitgtk-4.0/webkit2/WebKitJavascriptResult.h:52:1: note: declared here
 webkit_javascript_result_get_value          (WebKitJavascriptResult *js_result);
 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by leaanthony** [comment]
Not sure if you deleted your response but go modules turns off if in the gopath. Best got close to your home/kcode dir and use that.

**Comment by bh90210** [comment]
uh yes, cause I figured out that it was something I was doing wrong, didn't want to clatter the issue with that.

**Comment by bh90210** [comment]
https://github.com/wailsapp/wails/tree/92-build-on-first-init
this solves the issue yes.

**Comment by leaanthony** [comment]
Closed by #97

**Comment by boutheinaSymfony** [comment]
je résolu se problème de réinstallé npm cd "nmp install"
